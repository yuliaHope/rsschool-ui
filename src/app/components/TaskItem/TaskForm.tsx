import * as React from 'react';
import { IAssignmentModel } from 'core/models';

type Props = {
    status: string;
    assignment: IAssignmentModel;
    fetchAssignmentSolution: (assignment: IAssignmentModel) => void;
};

class TaskForm extends React.Component<Props> {
    private studentRepo: React.RefObject<HTMLInputElement>;
    private studentComment: React.RefObject<HTMLTextAreaElement>;
    constructor(props: Props) {
        super(props);
        this.studentRepo = React.createRef();
        this.studentComment = React.createRef();
    }

    SendStudentTask(assignment: IAssignmentModel) {
        const { fetchAssignmentSolution } = this.props;
        const studentRepoValue: string = this.studentRepo.current!.value.replace(/\s/g, '');
        if (studentRepoValue) {
            fetchAssignmentSolution(assignment);
        } else {
            alert('wrong repo');
            this.studentRepo.current!.value = '';
        }
    }

    render() {
        const { status } = this.props;
        if (status !== 'MissedDeadline' && status !== 'Checked' && status !== 'ReadyForReview') {
            const { assignment } = this.props;
            return (
                <small className="text-muted">
                    <form
                        onSubmit={ev => {
                            ev.preventDefault();
                            this.SendStudentTask(assignment);
                        }}
                    >
                        <div className="form-group">
                            <label>Choose repo</label>
                            <input
                                className="form-control form-control-sm"
                                type="text"
                                ref={this.studentRepo}
                                placeholder="Enter link"
                                pattern="https?://github.com/.+"
                            />
                        </div>
                        <div className="form-group">
                            <label>Comments</label>
                            <textarea
                                className="form-control form-control-sm"
                                ref={this.studentComment}
                                placeholder="Write comments here"
                            />
                        </div>
                        <button type="submit" className="btn btn-primary btn-sm">
                            Submit
                        </button>
                    </form>
                </small>
            );
        } else {
            return <React.Fragment />;
        }
    }
}

export default TaskForm;
