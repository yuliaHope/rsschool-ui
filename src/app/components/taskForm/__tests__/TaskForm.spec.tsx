import { shallow } from 'enzyme';
import * as React from 'react';
import TaskForm from '../index';

describe('TaskForm', () => {
    it('renders correctly if no data', () => {
        const output = shallow(
            <TaskForm
                courseId={''}
                title={''}
                urlToDescription={''}
                status={''}
                taskId={0}
                score={0}
                submit={jest.fn()}
            />,
        );
        expect(output).toMatchSnapshot();
    });
});
