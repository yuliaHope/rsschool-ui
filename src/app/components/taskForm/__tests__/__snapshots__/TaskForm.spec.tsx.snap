// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm renders correctly if no data 1`] = `
<div
  className="card border-undefined mb-3"
>
  <div
    className="card-header bg-undefined"
  />
  <FooterForm
    title=""
    urlToDescription=""
  />
  <div
    className="card-footer"
  />
</div>
`;
